eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -815,18 +837,23 @@ object Router {
 
   def props(nodeParams: NodeParams, watcher: ActorRef, initialized: Option[Promise[Done]] = None) = Props(new Router(nodeParams, watcher, initialized))
 
-  def toFakeUpdate(extraHop: ExtraHop): ChannelUpdate =
-  // the `direction` bit in flags will not be accurate but it doesn't matter because it is not used
-  // what matters is that the `disable` bit is 0 so that this update doesn't get filtered out
-    ChannelUpdate(signature = ByteVector64.Zeroes, chainHash = ByteVector32.Zeroes, extraHop.shortChannelId, Platform.currentTime.milliseconds.toSeconds, messageFlags = 0, channelFlags = 0, extraHop.cltvExpiryDelta, htlcMinimumMsat = 0 msat, extraHop.feeBase, extraHop.feeProportionalMillionths, None)
-
+  def toFakeUpdate(extraHop: ExtraHop, htlcMaximum: MilliSatoshi): ChannelUpdate = {
+    // the `direction` bit in flags will not be accurate but it doesn't matter because it is not used
+    // what matters is that the `disable` bit is 0 so that this update doesn't get filtered out
+    ChannelUpdate(signature = ByteVector64.Zeroes, chainHash = ByteVector32.Zeroes, extraHop.shortChannelId, Platform.currentTime.milliseconds.toSeconds, messageFlags = 1, channelFlags = 0, extraHop.cltvExpiryDelta, htlcMinimumMsat = 0 msat, extraHop.feeBase, extraHop.feeProportionalMillionths, Some(htlcMaximum))
+  }
 
-  def toAssistedChannels(extraRoute: Seq[ExtraHop], targetNodeId: PublicKey): Map[ShortChannelId, AssistedChannel] = {
+  def toAssistedChannels(extraRoute: Seq[ExtraHop], targetNodeId: PublicKey, amount: MilliSatoshi): Map[ShortChannelId, AssistedChannel] = {
     // BOLT 11: "For each entry, the pubkey is the node ID of the start of the channel", and the last node is the destination
```

Comment:
What's the advantage in computing the `htlcMaximumMsat` ourselves? The path finding would work fine even if there is no `htlcMaximumMsat`.

Reply:
Yes but if there's no `htlcMaximumMsat` the path finding uses the default lower bound, which penalizes the channel whereas we'd like to select those as much as possible since they were explicitly added to the invoice